packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -430,13 +525,41 @@ export class AnchorSet {
 		let parent: UpPath | undefined;
 		const moveTable = new Map<Delta.MoveId, UpPath>();
 
-		const visitor = {
+		// Run `withNode` on anchorNode for parent if there is such an anchorNode.
+		// If at root, run `withRoot` instead.
+		const maybeWithNode: (
+			withNode: (anchorNode: PathNode) => void,
+			withRoot?: () => void,
+		) => void = (withNode, withRoot) => {
+			if (parent === undefined && withRoot !== undefined) {
+				withRoot();
+				return;
+			}
+			assert(parent !== undefined, "parent must exist");
+			// TODO:Perf:
+			// When traversing to a depth D when there are not anchors in that subtree, this goes O(D^2).
+			// Delta traversal should early out in this case because no work is needed (and all move outs are known to not contain anchors).
+			parent = this.internalizePath(parent);
```

Comment:
This seems to break "can rebase over move". This is likely because the implementation of move here holds onto paths across when those paths could be edited, and this causes them to get partly updated.

This will likely be fixed by changing the implementation of move, which will also fix a bug related to how move out doesn't modify the tree when it should.

Reply:
broken test now has a skip to prevent that bug from blocking this work.